WordPress/gutenberg
# Update placeholder text color for standard themes

## What?
Fixes placeholder text colors for standard themes.

* https://github.com/wordpress-mobile/gutenberg-mobile/issues/6069

## Why?
Currently for sites using a standard theme like Russell, there is no distinction between a text input's primary color and its placeholder color.

## How?
Removes the usage of the `globalStyles.text` color as also functioning as the placeholder text color.

## Testing Instructions
1. Open a post using a site with a standard theme
2. Observe the Post Title and Paragraph placeholder colors are distinct from the primary text input color.

## Screenshots or screencast 
Before (trunk) | After (rnmobile/update-standard-theme-placeholder-text)
-|-
<img src="https://github.com/WordPress/gutenberg/assets/643285/59b51667-8c2a-4ec7-b7b9-6ea034f87893" width="350" /> | <img src="https://github.com/WordPress/gutenberg/assets/643285/1fe0ab67-79ec-4b03-a249-13157f1b2272" width="350" />
